### PR TITLE
Increase robustness in Gen3 updates

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

File: src/gateway/update_controller.py [hunk 1]
```diff
@@ -513,14 +519,19 @@ def _execute_pending_updates(self):
...
+                                                                   mode=UpdateEnums.Modes.AUTOMATIC,
+                                                                   module_address=None,
+                                                                   metadata=metadata)
+                had_work |= successes > 0 or failures > 0
             except Exception as ex:
                 component_logger.error('Could not update {0} to {1}: {2}'.format(firmware_type, target_version, ex))
 
+        if not had_work and self._pending_updates:
+            global_logger.info('No pending updates')
+            self._pending_updates = False
```
Comment (ptuytsch) [inline src/gateway/update_controller.py:533]: Correct me if i;m wrong, but I cannot see what is the purpose of this flag? This is never used to check something or if ti continute on updating pending updates?

File: src/master/core/slave_updater.py [hunk 2]
```diff
@@ -138,22 +142,17 @@ def update(address, hex_filename, gen3_firmware, version, logger, slave_communic
...
-                        break
-                    except CommunicationTimedOutException as ex:
-                        logger.warning('Flashing... Block {0} failed: {1}'.format(block, ex))
-                        if tries >= 3:
-                            raise
+                try:
+                    response = slave_communicator.do_command(address=address,
+                                                             command=SlaveAPI.write_firmware_block(),
+                                                             fields={'address': block, 'payload': payload},
+                                                             timeout=SlaveUpdater.WRITE_FLASH_BLOCK_TIMEOUT)
```
Comment (ptuytsch) [inline src/master/core/slave_updater.py:149]: Is the removal of the 3 retries on purpose here?

File: src/gateway/update_controller.py (hunk 1 above)
Comment (khenderick) [reply src/gateway/update_controller.py:533]: The update loop runs every minute, this flag is used to make sure the `No pending updates` is only printed once, and not every minute.

File: src/master/core/slave_updater.py (hunk 2 above)
Comment (khenderick) [reply src/master/core/slave_updater.py:149]: Yes, every `do_command` on the slave/ucan communicator now tries 3 times by default.

**Review by ptuytsch — APPROVED** [review]

**Pull request merged**